### DISCUSSION TITLE
Using `watch`, WASM file is sometimes served truncated

### DISCUSSION THREAD

**Discussion by hubertp** [enso-org/enso#5788] [Work In Progress - GUI] [answered]
Recently (~24.02) started getting a spurious error in IDE on startup (won't even boot the list of projects):
```
Uncaught (in promise) CompileError: WebAssembly.instantiateStreaming(): section (code 10, "Code") extends past end of the module (length 29962350, remaining bytes 27208289) @+185753
```

A recompilation of GUI does the trick but it is quite annoying as there is no apparent rule on when it happens.
Any ideas?
![Screenshot from 2023-03-01 17-20-48](https://user-images.githubusercontent.com/292128/222199675-69baf6de-4aa5-4177-8146-ecd0463055b3.png)

**Comment by kazcw** [comment]
Is the file truncated on disk, or are we downloading it incompletely--in a bad build, what's the size of `dist/gui/assets/pkg-opt.wasm`?

**Reply by hubertp** [reply]
The command I use to build it is pretty much `./run gui watch --skip-wasm-opt`. I can get you that data next time it happens

**Reply by hubertp** [reply]
```
> ls -la dist/wasm/pkg*wasm
-rw-rw-r-- 1 hubert hubert 60077987 Mar  2 11:05 dist/wasm/pkg-opt.wasm
-rw-rw-r-- 1 hubert hubert 60077987 Mar  2 11:05 dist/wasm/pkg.wasm
```

**Reply by hubertp** [reply]
Note that reloading the page doesn't help. I have to re-run `run` command.

**Reply by kazcw** [reply]
~~Ok, it's definitely a build problem, probably specific to `watch`~~

**Reply by wdanilo** [reply]
We've observed it couple weeks ago already. Our diagnostics back then was that indeed the generated WASM file was OK but it was not SERVED correctly. Reloading the page does not fix that, as it seems that the server caches the truncated file. I was able to observe that together with @mwu-tow when the file was being served while being re-computed. Thus, stopping the watch and running it again was fixing the issue. @hubertp could you please confirm that it is also fixed on your end if you just re-run the watch command without any changes to the code, please?

Anyway, this needs to be investigated and it is a mystery for us why it started happening.

**Reply by hubertp** [reply]
@wdanilo yeap, running it again was what I was doing as well.

**Comment by wdanilo** [comment, marked as answer]
@hubertp would you be so nice and create issue for it, please? I think ti could be labeled as `good first issue` as I believe that people without a lot of insights into our codebase could try resolving it.

**Reply by hubertp** [reply]
Sure, reported under https://github.com/enso-org/enso/issues/5860

**Reply by wdanilo** [reply]
Thank you!
